pkg/registry/apis/provisioning/repository/github.go

```diff
@@ -120,6 +120,14 @@ func (r *githubRepository) Read(ctx context.Context, logger *slog.Logger, filePa
 }
 
 func (r *githubRepository) Create(ctx context.Context, logger *slog.Logger, path, ref string, data []byte, comment string) error {
+	if ref == "" {
+		ref = r.config.Spec.GitHub.Branch
+	} else {
+		if err := r.createBranch(ctx, ref); err != nil {
```

Comment:
is it always create branch?  or more like "make sure branch exists"?

What happens if you pass in a git hash as a branch name 🤔

Reply:
> is it always create branch? or more like "make sure branch exists"?

I was thinking in "always create a new branch". 

My rationale is the following: 
- When users will open the dashboard out of the configured branch. When they click save, the dashboard created will be based on that **point of reference**. If you push that to other branch it may lose some meaning. Another branch could have a totally different version of that dashboard or nothing at all. Consequently, the delta may not be what they apply but maybe more. 
- If users apply changes based on branch into another branch, it's likely they will 
- I think it's dangerous to push to existing branches. I think it's way safer to create a branch from the reference point with a little delta. 
- If we don't always create a new branch, the UI would lead users to create a PR on each save on that branch (if we don't prevent it). 

This was my rationale but product may have a different opinion on this. 



> What happens if you pass in a git hash as a branch name 🤔

If you push the entire hash, it fails: 
```json
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "create branch on update: create branch: create branch ref: POST https://api.github.com/repos/grafana/git-ui-sync-demo/git/refs: 422 Sorry, branch or tag names consisting of 40 hex characters are not allowed. []",
  "code": 500
}
```

If you push with the cropped hash, it creates a branch with that name: 
![Screenshot 2024-11-25 at 20 04 43](https://github.com/user-attachments/assets/ce1c4709-d5e3-48de-8f7f-1eeae1889c4f)

Reply:
**offline discussion**: https://raintank-corp.slack.com/archives/C07V91MA04B/p1732561800080969

we agreed on letting us push to any branch so that the can use grafana UI to submit to a new version of the dashboard.
The comments Grafana will do on the PR will contain links to the previews which will be editable.
Those links will contain the ref (branch name or commit) and the link of the existing PR so that users can update it.  cc 
[@alex.khomenko](https://raintank-corp.slack.com/team/UT1873LP3)
. In that way,  the frontend will know and we wouldn't have to implement a complex logic to find if there is already an existing PR.
Users should be given the option to open a new PR.